python/dolfinx/wrappers/graph.cpp

```diff
@@ -32,6 +32,14 @@ void declare_adjacency_list(py::module& m, std::string type)
         std::vector<T> data(adj.data(), adj.data() + adj.size());
         return dolfinx::graph::build_adjacency_list<T>(std::move(data), dim);
       }))
+      .def(py::init(
+          [](const py::array_t<T, py::array::c_style>& array,
+             const py::array_t<std::int32_t, py::array::c_style>& displ) {
+            std::vector<T> data(array.data(), array.data() + array.size());
```

Comment:
Create the vectors directly in the AdjacencyList argument list or use std::move to void two copies.

Reply:
Changed to:
> ```c++
> std::vector data(array.data(), array.data() + array.size());
> std::vector offsets(displ.data(), displ.data() + displ.size());
> return dolfinx::graph::AdjacencyList<T>(std::move(data), std::move(offsets));
> ```